src/components/SideNavbar/SideNavbar.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+// eslint-disable-next-line sort-imports
+import SidenavMenu from './SidenavMenu';
+
+const Nav = styled.div`
+  width: 80px;
+  height: 1110px;
+  padding: 40px 24px 91px;
+
+  box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.15);
+  background-color: #fff;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const KeywordBlock = styled.div`
+  width: 32px;
+  height: 32px;
+  margin: 20px 0;
+  object-fit: contain;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const HistoryBlock = styled.div`
+  width: 32px;
+  height: 32px;
+  ${({ sideMenu }) => {
+    return sideMenu ? `margin :   289px 0 0 0 ` : `margin: 20px 0`;
+  }};
+  object-fit: contain;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const ChatBlock = styled.div`
+  width: 32px;
+  height: 32px;
+  ${({ sideMenu }) => {
+    return sideMenu ? `margin :   45px 0 0 0 ` : `margin: 20px 0`;
+  }};
+  object-fit: contain;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const SetBlock = styled.div`
+  width: 32px;
+  height: 32px;
+  ${({ sideMenu }) => {
+    return sideMenu ? `margin :   397px 0 0 0 ` : `margin: 20px 0`;
+  }};
+  object-fit: contain;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const OpenMenu = styled.div`
+  width: 26px;
+  height: 20px;
+  margin: 0 3px 60px;
+  object-fit: contain;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const MenuButton = styled.button`
+  width: 26px;
+  height: 20px;
+  cursor: pointer;
+  object-fit: contain;
+  border: 0;
+  display: flex;
+  background-color: #fff;
+  justify-content: center;
+  align-items: center;
+`;
+
+const NavContainer = styled.div`
+  width: 395px;
+  height: 1110px;
+  display: flex;
+`;
+
+const SideNavbar = () => {
+  const [sideMenu, setsideMenu] = useState(false);
+
+  const showSideMenu = () => {
+    setsideMenu((current) => !current);
+  };
```

Comment:
showSideMenu 는 오직 sideMenu를 보여주게 만드는 것 같은데,
현재 코드는 이전 상태를 계속해서 바꾸는 것으로 보입니다.
현 함수 이름을 유지하려면 로직을 바꾸던가, 이름을 바꾸세요.

```suggestion
  const showSideMenu = () => setSideMenu(true);
```

```suggestion
  const toggleSideMenu = () => {
    setSideMenu(prev => !prev);
  };
```

Reply:
725609a 수정완료